graph_neural_network/train_rgnn_multi_gpu.py

```diff
@@ -0,0 +1,356 @@
+    input_nodes=('paper', val_idx),
+    batch_size=val_batch_size,
+    shuffle=True,
+    drop_last=False,
+    device=current_device,
+    seed=random_seed
+  )
+  # Load checkpoint
+  ckpt = None
+  if ckpt_path is not None:
+    try:
+      ckpt = torch.load(ckpt_path)
+    except FileNotFoundError as e:
+      print(f"Checkpoint file not found: {e}")
+      return -1
+    
+  # Define model and optimizer.
+  model = RGNN(dataset.get_edge_types(),
+               dataset.node_features['paper'].shape[1],
+               hidden_channels,
+               num_classes,
+               num_layers=num_layers,
+               dropout=0.2,
+               model=model_type,
+               heads=num_heads,
+               node_type='paper').to(current_device)
+  if ckpt is not None:
+    model.load_state_dict(ckpt['model_state_dict'])
+  model = DistributedDataParallel(model,
+                                  device_ids=[current_device.index] if with_gpu else None,
+                                  find_unused_parameters=True)
+
+  param_size = 0
+  for param in model.parameters():
+    param_size += param.nelement() * param.element_size()
+  buffer_size = 0
+  for buffer in model.buffers():
+    buffer_size += buffer.nelement() * buffer.element_size()
+
+  size_all_mb = (param_size + buffer_size) / 1024**2
+  print('model size: {:.3f}MB'.format(size_all_mb))
+
+  loss_fcn = torch.nn.CrossEntropyLoss().to(current_device)
+  optimizer = torch.optim.Adam(model.parameters(), lr=learning_rate)
+  if ckpt is not None:
+    optimizer.load_state_dict(ckpt['optimizer_state_dict'])
+
+  batch_num = (len(train_idx) + train_batch_size - 1) // train_batch_size
+  validation_freq = int(batch_num * validation_frac_within_epoch)
+  is_success = False
+  epoch_num = 0
+
+  training_start = time.time()
+  for epoch in tqdm.tqdm(range(epochs)):
+    model.train()
+    total_loss = 0
+    train_acc = 0
+    idx = 0
+    gpu_mem_alloc = 0
+    epoch_start = time.time()
+    for batch in train_loader:
+      idx += 1
+      batch_size = batch['paper'].batch_size
+      out = model(
+        {
+          node_name: node_feat.to(current_device).to(torch.float32)
+          for node_name, node_feat in batch.x_dict.items()
+        },  
+        batch.edge_index_dict
+      )[:batch_size]
+      y = batch['paper'].y[:batch_size]
+      loss = loss_fcn(out, y)
+      optimizer.zero_grad()
+      loss.backward()
+      optimizer.step()
+      total_loss += loss.item()
+      train_acc += sklearn.metrics.accuracy_score(y.cpu().numpy(),
+          out.argmax(1).detach().cpu().numpy())*100
+      gpu_mem_alloc += (
+          torch.cuda.max_memory_allocated() / 1000000
+          if with_gpu
+          else 0
+      )
+      #checkpoint
+      if ckpt_steps > 0 and idx % ckpt_steps == 0:
+        if with_gpu:
+          torch.cuda.synchronize()
+        dist.barrier()
+        if rank == 0:
+          epoch_num = epoch + idx / batch_num
```

Comment:
Seems like the reported epoch number is not rounded here to 2 significant digits. Could we round it up here so that we can be consistent with the [RCP PR](https://github.com/mlcommons/logging/pull/359/files)?

Reply:
fixed